packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -29,19 +39,214 @@ import DefaultStylePicker from '../default-style-picker';
+	const isContentBlock = useCallback(
+		( blockName ) => {
+			return !! contenBlocksObjectAux[ blockName ];
+		},
+		[ blockTypes ]
+	);
+	return useMemo( () => {
+		return getContentBlocks( [ block ], isContentBlock );
+	}, [ block, isContentBlock ] );
+}
+
+function getContentBlocks( blocks, isContentBlock ) {
+	const result = [];
+	for ( const block of blocks ) {
+		if ( isContentBlock( block.name ) ) {
+			result.push( block );
+		}
+		result.push( ...getContentBlocks( block.innerBlocks, isContentBlock ) );
+	}
+	return result;
+}
+
+function BlockInspectorNavigationEffects( { children } ) {
+	const { selectBlock } = useDispatch( blockEditorStore );
+	const { goTo, location } = useNavigator();
+	const lastLocationClientId = useRef();
+	const updatingSelectionTo = useRef();
+	const locationClientId = useMemo( () => {
+		if ( location.path.startsWith( '/block/' ) ) {
+			return location.path.substring( '/block/'.length );
+		}
+	}, [ location.path ] );
+	const selectedBlockId = useSelect( ( select ) => {
+		return select( blockEditorStore ).getSelectedBlockClientId();
+	} );
+	// When the location changes update the selection to match the new location.
+	useEffect( () => {
+		lastLocationClientId.current = locationClientId;
+		if ( locationClientId && selectedBlockId !== locationClientId ) {
+			updatingSelectionTo.current = locationClientId;
+			selectBlock( locationClientId );
+		}
+	}, [ locationClientId ] );
+	// When the selection changes update the location to root if no selection update to match location is in progress.
+	useEffect( () => {
+		if ( updatingSelectionTo.current ) {
+			updatingSelectionTo.current = undefined;
+		} else if ( lastLocationClientId.current ) {
+			goTo( '/' );
+		}
+	}, [ selectedBlockId ] );
+
+	return children;
+}
+
+function BlockNavigationButton( { blockTypes, block, selectedBlock } ) {
+	const blockType = blockTypes.find( ( { name } ) => name === block.name );
+	return (
+		<NavigatorButton
+			path={ `/block/${ block.clientId }` }
+			style={
+				selectedBlock.clientId === block.clientId
+					? { color: 'var(--wp-admin-theme-color)' }
+					: {}
+			}
+		>
+			<HStack justify="flex-start">
+				<BlockIcon icon={ blockType.icon } />
+				<FlexItem>{ blockType.title }</FlexItem>
+			</HStack>
+		</NavigatorButton>
+	);
+}
+
+function BlockNavigatorScreen( { block } ) {
+	return (
+		<NavigatorScreen path={ `/block/${ block.clientId }` }>
+			<BlockInspectorSingleBlock
+				clientId={ block.clientId }
+				blockName={ block.name }
+				backButton={
+					<NavigatorBackButton
+						style={
+							// TODO: This style override is also used in ToolsPanelHeader.
+							// It should be supported out-of-the-box by Button.
+							{ minWidth: 24, padding: 0 }
+						}
+						icon={ isRTL() ? chevronRight : chevronLeft }
+						isSmall
+						aria-label={ __( 'Navigate to the previous view' ) }
+					/>
+				}
+			/>
+		</NavigatorScreen>
+	);
+}
+
+function BlockInspectorAbsorvedBy( { absorvedBy } ) {
```

Comment:
I've had troubles to parse what `BlockInspectorAbsorvedBy` and `absorvedBy` meant. I'm not sure about the right naming here and wanted to share some ideas to see if we can come up with a better naming schema: `BlockInspectorLockedBlocks` and `topLevelLockedBlock`?